File: PHP/Disk.php
<?php
    class Disk
    {
        // Disk attributes.
        private $type;
        private $capacity;
        private $price;
        
        /* Constructor.
           Please take note that after a few try, PHP seems to not accept
           multiple constructor. So I decided to set it as data constructor. */
        
        // Constructor with disk's data.
        public function __construct($type, $capacity, $price)
        {
            $this->type = $type;
            $this->capacity = $capacity;
            $this->price = $price;            
        }

        /* Setter and Getter. */

        // Set type attribute.
        public function setType($type)
        {
            $this->type = $type;
        }

        // Set capacity attribute.
        public function setCapacity($capacity)
        {
            $this->capacity = $capacity;
        }

        // Set price attribute.
        public function setPrice($price)
        {
            $this->price = $price;
        }

        // Get type attribute.
        public function getType()
        {
            return $this->type;
        }

        // Get capacity attribute.
        public function getCapacity()
        {
            return $this->capacity;
        }

        // Get price attribute.
        public function getPrice()
        {
            return $this->price;
        }

        /* Destructor. */

        public function __destruct()
        {
            
        }
    }
?>

File: C++/Disk.cpp
#include "header.hh"

/* Constructor. */

// Empty constructor.
Disk::Disk()
{

}

// Constructor with disk's data.
Disk::Disk(string type, int capacity, int price)
{
    this->type = type;
    this->capacity = capacity;
    this->price = price;
}

/* Setter and Getter. */

// Set type attribute.
void Disk::setType(string type)
{
    this->type = type;
}

// Set capacity attribute.
void Disk::setCapacity(int capacity)
{
    this->capacity = capacity;
}

// Set price attribute.
void Disk::setPrice(int price)
{
    this->price = price;
}

// Get type attribute.
string Disk::getType()
{
    return type;
}

// Get capacity attribute.
int Disk::getCapacity()
{
    return capacity;
}

// Get price attribute.
int Disk::getPrice()
{
    return price;
}

/* Destructor. */

Disk::~Disk()
{
    
}